Documentation/admission-controller-usage.md

```diff
@@ -0,0 +1,51 @@
+---
+title: Admission Controller
+weight: 2030
+indent: true
+---
+
+# Admission Controller
+
+An admission controller intercepts requests to the Kubernetes API server prior to persistence of the object, but after the request is authenticated and authorized.
+
+Enabling the Rook admission controller is recommended to provide an additional level of validation that Rook is configured correctly with the custom resource (CR) settings.
+
+## Quick Start 
+
+To deploy the Rook admission controllers we have a helper script that will automate the configuration.
+
+This script will help us achieve the following tasks
+1. Creates self-signed certificate.
+1. Creates a CSR for the certificate and gets it approved from the kubernetes cluster.
+1. Stores these certificates as a secret.
+1. Creates a service account, clusterrole and clusterrolebindings for running the webhook server with minimal priviledges.
+1. Creates ValidatingWebhookConfig and fills the CA bundle with the appropriate value from the cluster.
+
+Run the following commands
+```
+kubectl create -f examples/kubernetes/ceph/common.yaml
+cluster/examples/kubernetes/ceph/admission-controller/deploy.sh
+```
+Now that the secrets have been deployed, we can deploy the operator and create a cluster using the following commands
+```
+kubectl create -f operator.yaml
+```
+
+At this point the operator will start the admission controller deployment automatically and the webhook will start intercepting requests for Rook resources. 
+
+## Certificate Management
```

Comment:
Why not just use a `cert-manager`? rather than creating a certificate using shell script, we can create a certificate using `cert-manager` and add `cert-manager.io/inject-ca-from` annotation in `ValidatingWebhookConfiguration`

Reply:
@prksu I am not very well versed with the usage of cert-manager. Can you expand upon it and provide any links ?

Reply:
we can create the certificate like this
```yaml
apiVersion: cert-manager.io/v1alpha2
kind: Issuer
metadata:
  name: selfsigned-issuer
  namespace: system
spec:
  selfSigned: {}
---
apiVersion: cert-manager.io/v1alpha2
kind: Certificate
metadata:
  name: serving-cert
  namespace: system
spec:
  dnsNames:
  - $(SERVICE_NAME).$(SERVICE_NAMESPACE).svc
  - $(SERVICE_NAME).$(SERVICE_NAMESPACE).svc.cluster.local
  issuerRef:
    kind: Issuer
    name: selfsigned-issuer
  secretName: webhook-server-cert
```

and the `ValidatingWebhookConfiguration` will looks like

```yaml
apiVersion: admissionregistration.k8s.io/v1beta1
kind: ValidatingWebhookConfiguration
metadata:
  name: demo-webhook
  annotations:
    cert-manager.io/inject-ca-from: $(CERTIFICATE_NAMESPACE)/$(CERTIFICATE_NAME)
webhooks:
  - name: webhook-server.webhook-demo.svc
    clientConfig:
      service:
        name: webhook-server
        namespace: rook-ceph
        path: "/validate"
      caBundle: Cg==
    rules:
      - operations: [ "CREATE" ]
        apiGroups: [""]
        apiVersions: ["v1"]
        resources: ["pods"]
```

the `cert-manager` will automatically replace `webhooks.clientConfig.caBundle` with appropriate `certificate`

ref:
https://cert-manager.io/docs/configuration/selfsigned/
https://cert-manager.io/docs/concepts/ca-injector/